From ad75deb2aba983f9226f18bec62c84cc021a2cd8 Mon Sep 17 00:00:00 2001
From: Hidehiko Abe <hidehiko@chromium.org>
Date: Tue, 4 Sep 2018 17:51:30 +0900
Subject: [PATCH] [PATCH] Manually cherrypick
 ScopedClosureRunner::ReplaceClosure().

https://codereview.chromium.org/2094803005 replaces
ScopedClosureRunner::Reset(). This CL cherry-picks the new methods
from the CL so that the uprev work can be step-by-step.

BUG=b:37434548
TEST=Trybot.

Change-Id: Id25af5556fdf36dfd9039cdb804e5ce6a35dcff2
---
 base/callback_helpers.cc | 4 ++++
 base/callback_helpers.h  | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/base/callback_helpers.cc b/base/callback_helpers.cc
index ef02b2b..82ac435 100644
--- a/base/callback_helpers.cc
+++ b/base/callback_helpers.cc
@@ -39,4 +39,8 @@ Closure ScopedClosureRunner::Release() {
   return result;
 }
 
+void ScopedClosureRunner::ReplaceClosure(const Closure& closure) {
+  closure_ = closure;
+}
+
 }  // namespace base
diff --git a/base/callback_helpers.h b/base/callback_helpers.h
index 8608039..a6e9c39 100644
--- a/base/callback_helpers.h
+++ b/base/callback_helpers.h
@@ -39,6 +39,11 @@ class BASE_EXPORT ScopedClosureRunner {
   void Reset(const Closure& closure);
   Closure Release() WARN_UNUSED_RESULT;
 
+  // Manual cherrypick from https://codereview.chromium.org/2094803005
+  // for libchrome uprev.
+  void ResetAndRun() { Reset(); }
+  void ReplaceClosure(const Closure& closuer);
+
  private:
   Closure closure_;
 
-- 
2.19.0.rc1.350.ge57e33dbd1-goog

